Improve sentences
----------------------------

In this notebook we want to improve the corpora of new sentences and the final corpora which is the concatenation of the original corpora and the new one (we got the two in the following notebook [new_sentences](extract_new_sentences.ipynb)). 

For the improvement we want to make the following modifications that we found to be necessary after fine-tuning the `t5` model on the new sentences:

- Remove the definitions: We had already extract the new sentences without the definitions we will only need to load the "true sentences".
- Replace exclamation marks (!) at the end of the phrases by period marks (.): This will help the transformers to identify the end of a sentence. Not that we will do this for both of the new sentences and the last version corpora (v4). We cannot know if it is the right thing to do in order to obtain better performance that's why we will create a function and add it when making hyper-parameter search and reduce the search space if necessary.
- Remove sentences with less than 3 tokens in one of the French of Wolof version. This will reduce the number of sentences but can provide better agreement with the right way to write sentences.

The above modifications on the version 4 of the corpora and the diagne's sentences will be necessary to create the version 5 of the corpora. The additional version of the diagne's sentences and 6 of the corpora will understand additionally sentences got from `omniglot`. We will not select the sentences by length for the new versions.

All the modifications can be done with the `pandas` library.

In [1]:
import re
import pandas as pd
from tokenizers.pre_tokenizers import Whitespace
from wolof_translate.utils.database_manager import TranslationMongoDBManager

### Remove the definitions

Let us load the two corpora.

In [2]:
corpora_v4 = pd.read_csv('data/extractions/new_data/corpora_v4.csv')

diagne_sentences = pd.read_csv('data/extractions/new_data/sentences.csv')

In [3]:
corpora_v4.to_csv('verification.csv', index=False)

In [4]:
corpora_v4.shape[0], diagne_sentences.shape[0]

(2500, 1615)

### Add periods to the end of the sentences

We have three possibilities:

- We can replace the exclamations by period marks (option 2)
- ... add period mark at each end of sentences (option 3)
- ... do the two previous modification at same time (option 4)

The following function will be added inside the `TransformerSequences` class when recuperating the sentences before the training.

In [5]:
%%writefile wolof-translate/wolof_translate/utils/improvements/end_marks.py

import pandas as pd
from typing import *

def add_end_mark(sentences: Union[list, str], end_mark: str = ".", end_mark_to_remove: Union[str, None] = None, 
                 replace: bool = False, poss_end_marks: list = ['!', '?', '.', '...']):
    
    if isinstance(sentences, str): sentences = [sentences]
    
    if replace and end_mark_to_remove is None:
        
        raise ValueError("You must provide a end mark to remove if you want to make replacement !")
    
    new_sentences = []
    
    # if replace is chosen we will replace the end to remove by the end mark if not only add end mark to each sentence and remove the end mark to remove
    for sentence in sentences:
        
        if replace:
            
            if sentence[-1] == end_mark_to_remove:
                
                sentence = sentence[:-1].strip() + end_mark
            
        else:
            
            if not end_mark_to_remove is None and sentence[-1] == end_mark_to_remove:
                
                sentence = sentence[:-1]
            
            if not sentence[-1] in poss_end_marks: 
                
                sentence = sentence.strip() + end_mark
        
        new_sentences.append(sentence)
    
    return new_sentences
    

Overwriting wolof-translate/wolof_translate/utils/improvements/end_marks.py


Let us try the different type of end mark adding with the above function.

In [6]:
%run wolof-translate/wolof_translate/utils/improvements/end_marks.py

In [7]:
# recuperate some sentences from the diagne's sentences
sample = diagne_sentences.sample(30)

In [8]:
sample

french  \
1010                Et ne viens surtout pas à l'école !   
104                             Ce mouton-là en entier.   
284                              C'est quand tu es venu   
1531    Si tu veux, retourne au centre de Saint-Louis !   
204                                   C'est des femmes.   
1564  Je veux que tu viennes le matin à cinq heures ...   
317                                             Vas-y !   
1560       Ce matin ou ce soir c'est moi qui suis ici !   
719                                     C'est celui-là.   
54                                J'ai aperçu un homme.   
985                          Là où il y a de la lumière   
20                                    J'ai vu la femme.   
124                        Aucun mouton ne s'est égaré.   
470             Les gens n'eussent pas été des maîtres.   
1336                       J'ai vu des vaches d'Europe.   
1482  Les enfants ont cherché à l'intérieur, à l'ext...   
1326                                  Vous fûtes Laobé.   
1440         L'homme a tenu aux enfants de bons propos.   
872                         Il est venu à cause de toi.   
45                     Je connaissais cet individu-là !   
4                             Homme n'est pas mauvais !   
697                          D'ailleurs, je m'en vais !   
1265                       J'ai aperçu une jeune fille.   
1340   Il n'est de Ardo du Dieri que tu ne connaisses !   
731                                   Que pleure-t-il ?   
1284                                Honore tout homme !   
663                              Voilà ce qu'il en est.   
1495        L'enfant dit d'entrer pour qu'il vous voie.   
649                                              Quoi ?   
248                        C'est cette ville de Sérère.   

                                               wolof  
1010                        Te bul ñëw jaŋgu ba de !  
104                                Xar moomule mépp.  
284                                  Bi ŋga ñëwée la  
1531                  Dellu, soo bëggée, biir Ndar !  
204                                     Jigéen lanu.  
1564   Bëgg naa ci juróomi waxtu, ci subë, ŋga ñëw !  
317                                         Dem fa !  
1560        Maa fiy nekk tày ci ŋgoon, tày ci suba !  
719                                         Kile la.  
54                                  Séen naa aw nit.  
985                                 Ci biir fiy leer  
20                                Gis naa jigéen ju.  
124                                 Menn xar réerul.  
470                      Nit ñi doonkoonuñu ay kaaŋ.  
1336                             Gis naa nagu Tugël.  
1482  Xale yi set nañu ci biir, ci biti gisuñu dara.  
1326                               Laobe ŋgeen woon.  
1440            Góor gi wax na xale yi wax ju rafet.  
872                                Dikk na ndax yaw.  
45                           Xamoon naa waa joojee !  
4                                        Nit bonul !  
697                            Te sax, ma ŋgii dem !  
1265                               Séen naa as ndaw.  
1340               Amul benn ardow Jéeri boo xamul !  
731                                    Lu muy jooy ?  
1284                              Teralal bépp nit !  
663                                        Loolu la.  
1495              Xale bi ne ŋgeen dugg mu gis leen.  
649                                        Lii lan ?  
248                            Gii dëkku Séeréer la.

We need to add the modification on each version of the sentences.

- Replace exclamations by end marks.

In [9]:
french_sents = sample['french'].to_list()

wolof_sents = sample['wolof'].to_list()

french_sents = add_end_mark(french_sents, end_mark_to_remove='!', replace=True)

wolof_sents = add_end_mark(wolof_sents, end_mark_to_remove='!', replace=True)

pd.DataFrame({'french': french_sents, 'wolof': wolof_sents})


french  \
0                  Et ne viens surtout pas à l'école.   
1                             Ce mouton-là en entier.   
2                              C'est quand tu es venu   
3      Si tu veux, retourne au centre de Saint-Louis.   
4                                   C'est des femmes.   
5   Je veux que tu viennes le matin à cinq heures ...   
6                                              Vas-y.   
7         Ce matin ou ce soir c'est moi qui suis ici.   
8                                     C'est celui-là.   
9                               J'ai aperçu un homme.   
10                         Là où il y a de la lumière   
11                                  J'ai vu la femme.   
12                       Aucun mouton ne s'est égaré.   
13            Les gens n'eussent pas été des maîtres.   
14                       J'ai vu des vaches d'Europe.   
15  Les enfants ont cherché à l'intérieur, à l'ext...   
16                                  Vous fûtes Laobé.   
17         L'homme a tenu aux enfants de bons propos.   
18                        Il est venu à cause de toi.   
19                    Je connaissais cet individu-là.   
20                           Homme n'est pas mauvais.   
21                          D'ailleurs, je m'en vais.   
22                       J'ai aperçu une jeune fille.   
23    Il n'est de Ardo du Dieri que tu ne connaisses.   
24                                  Que pleure-t-il ?   
25                                 Honore tout homme.   
26                             Voilà ce qu'il en est.   
27        L'enfant dit d'entrer pour qu'il vous voie.   
28                                             Quoi ?   
29                       C'est cette ville de Sérère.   

                                             wolof  
0                          Te bul ñëw jaŋgu ba de.  
1                                Xar moomule mépp.  
2                                  Bi ŋga ñëwée la  
3                    Dellu, soo bëggée, biir Ndar.  
4                                     Jigéen lanu.  
5     Bëgg naa ci juróomi waxtu, ci subë, ŋga ñëw.  
6                                          Dem fa.  
7          Maa fiy nekk tày ci ŋgoon, tày ci suba.  
8                                         Kile la.  
9                                 Séen naa aw nit.  
10                                Ci biir fiy leer  
11                              Gis naa jigéen ju.  
12                                Menn xar réerul.  
13                     Nit ñi doonkoonuñu ay kaaŋ.  
14                             Gis naa nagu Tugël.  
15  Xale yi set nañu ci biir, ci biti gisuñu dara.  
16                               Laobe ŋgeen woon.  
17            Góor gi wax na xale yi wax ju rafet.  
18                               Dikk na ndax yaw.  
19                          Xamoon naa waa joojee.  
20                                      Nit bonul.  
21                            Te sax, ma ŋgii dem.  
22                               Séen naa as ndaw.  
23                Amul benn ardow Jéeri boo xamul.  
24                                   Lu muy jooy ?  
25                               Teralal bépp nit.  
26                                       Loolu la.  
27              Xale bi ne ŋgeen dugg mu gis leen.  
28                                       Lii lan ?  
29                           Gii dëkku Séeréer la.

- Add end marks at each sentence without replacing or removing

In [10]:
french_sents = sample['french'].to_list()

wolof_sents = sample['wolof'].to_list()

french_sents = add_end_mark(french_sents)

wolof_sents = add_end_mark(wolof_sents)

pd.DataFrame({'french': french_sents, 'wolof': wolof_sents})


french  \
0                 Et ne viens surtout pas à l'école !   
1                             Ce mouton-là en entier.   
2                             C'est quand tu es venu.   
3     Si tu veux, retourne au centre de Saint-Louis !   
4                                   C'est des femmes.   
5   Je veux que tu viennes le matin à cinq heures ...   
6                                             Vas-y !   
7        Ce matin ou ce soir c'est moi qui suis ici !   
8                                     C'est celui-là.   
9                               J'ai aperçu un homme.   
10                        Là où il y a de la lumière.   
11                                  J'ai vu la femme.   
12                       Aucun mouton ne s'est égaré.   
13            Les gens n'eussent pas été des maîtres.   
14                       J'ai vu des vaches d'Europe.   
15  Les enfants ont cherché à l'intérieur, à l'ext...   
16                                  Vous fûtes Laobé.   
17         L'homme a tenu aux enfants de bons propos.   
18                        Il est venu à cause de toi.   
19                   Je connaissais cet individu-là !   
20                          Homme n'est pas mauvais !   
21                         D'ailleurs, je m'en vais !   
22                       J'ai aperçu une jeune fille.   
23   Il n'est de Ardo du Dieri que tu ne connaisses !   
24                                  Que pleure-t-il ?   
25                                Honore tout homme !   
26                             Voilà ce qu'il en est.   
27        L'enfant dit d'entrer pour qu'il vous voie.   
28                                             Quoi ?   
29                       C'est cette ville de Sérère.   

                                             wolof  
0                         Te bul ñëw jaŋgu ba de !  
1                                Xar moomule mépp.  
2                                 Bi ŋga ñëwée la.  
3                   Dellu, soo bëggée, biir Ndar !  
4                                     Jigéen lanu.  
5    Bëgg naa ci juróomi waxtu, ci subë, ŋga ñëw !  
6                                         Dem fa !  
7         Maa fiy nekk tày ci ŋgoon, tày ci suba !  
8                                         Kile la.  
9                                 Séen naa aw nit.  
10                               Ci biir fiy leer.  
11                              Gis naa jigéen ju.  
12                                Menn xar réerul.  
13                     Nit ñi doonkoonuñu ay kaaŋ.  
14                             Gis naa nagu Tugël.  
15  Xale yi set nañu ci biir, ci biti gisuñu dara.  
16                               Laobe ŋgeen woon.  
17            Góor gi wax na xale yi wax ju rafet.  
18                               Dikk na ndax yaw.  
19                         Xamoon naa waa joojee !  
20                                     Nit bonul !  
21                           Te sax, ma ŋgii dem !  
22                               Séen naa as ndaw.  
23               Amul benn ardow Jéeri boo xamul !  
24                                   Lu muy jooy ?  
25                              Teralal bépp nit !  
26                                       Loolu la.  
27              Xale bi ne ŋgeen dugg mu gis leen.  
28                                       Lii lan ?  
29                           Gii dëkku Séeréer la.

- Remove exclamation marks and add period marks

In [11]:
french_sents = sample['french'].to_list()

wolof_sents = sample['wolof'].to_list()

french_sents = add_end_mark(french_sents, end_mark_to_remove='!')

wolof_sents = add_end_mark(wolof_sents, end_mark_to_remove='!')

pd.DataFrame({'french': french_sents, 'wolof': wolof_sents})


french  \
0                  Et ne viens surtout pas à l'école.   
1                             Ce mouton-là en entier.   
2                             C'est quand tu es venu.   
3      Si tu veux, retourne au centre de Saint-Louis.   
4                                   C'est des femmes.   
5   Je veux que tu viennes le matin à cinq heures ...   
6                                              Vas-y.   
7         Ce matin ou ce soir c'est moi qui suis ici.   
8                                     C'est celui-là.   
9                               J'ai aperçu un homme.   
10                        Là où il y a de la lumière.   
11                                  J'ai vu la femme.   
12                       Aucun mouton ne s'est égaré.   
13            Les gens n'eussent pas été des maîtres.   
14                       J'ai vu des vaches d'Europe.   
15  Les enfants ont cherché à l'intérieur, à l'ext...   
16                                  Vous fûtes Laobé.   
17         L'homme a tenu aux enfants de bons propos.   
18                        Il est venu à cause de toi.   
19                    Je connaissais cet individu-là.   
20                           Homme n'est pas mauvais.   
21                          D'ailleurs, je m'en vais.   
22                       J'ai aperçu une jeune fille.   
23    Il n'est de Ardo du Dieri que tu ne connaisses.   
24                                  Que pleure-t-il ?   
25                                 Honore tout homme.   
26                             Voilà ce qu'il en est.   
27        L'enfant dit d'entrer pour qu'il vous voie.   
28                                             Quoi ?   
29                       C'est cette ville de Sérère.   

                                             wolof  
0                          Te bul ñëw jaŋgu ba de.  
1                                Xar moomule mépp.  
2                                 Bi ŋga ñëwée la.  
3                    Dellu, soo bëggée, biir Ndar.  
4                                     Jigéen lanu.  
5     Bëgg naa ci juróomi waxtu, ci subë, ŋga ñëw.  
6                                          Dem fa.  
7          Maa fiy nekk tày ci ŋgoon, tày ci suba.  
8                                         Kile la.  
9                                 Séen naa aw nit.  
10                               Ci biir fiy leer.  
11                              Gis naa jigéen ju.  
12                                Menn xar réerul.  
13                     Nit ñi doonkoonuñu ay kaaŋ.  
14                             Gis naa nagu Tugël.  
15  Xale yi set nañu ci biir, ci biti gisuñu dara.  
16                               Laobe ŋgeen woon.  
17            Góor gi wax na xale yi wax ju rafet.  
18                               Dikk na ndax yaw.  
19                          Xamoon naa waa joojee.  
20                                      Nit bonul.  
21                            Te sax, ma ŋgii dem.  
22                               Séen naa as ndaw.  
23                Amul benn ardow Jéeri boo xamul.  
24                                   Lu muy jooy ?  
25                               Teralal bépp nit.  
26                                       Loolu la.  
27              Xale bi ne ŋgeen dugg mu gis leen.  
28                                       Lii lan ?  
29                           Gii dëkku Séeréer la.

### Remove sentences with less than 3 tokens

Only the words interest us and we want to separate them by space or marks. Notice also that we want verify for each version and if one it is not accepted then the another one also.

In [12]:
def select_by_length(sentences: pd.DataFrame, min_len: int = 4):
    
    new_sentences = {'french': [], 'wolof': []}
    
    for i in list(sentences.index):
        
        sent1 = sentences.loc[i, 'french']
        
        sent2 = sentences.loc[i, 'wolof']
        
        tokens1 = re.findall('\w+', sent1)
        
        tokens2 = re.findall('\w+', sent2)
        
        take = True
        
        if len(tokens1) < min_len or len(tokens2) < min_len:
            
            take = False
        
        if take:
            
            new_sentences['french'].append(sent1)
            
            new_sentences['wolof'].append(sent2)
            
    return pd.DataFrame(new_sentences)

Let us make an example with the sample that we got earlier.

In [13]:
sample = select_by_length(sample, min_len=3)


In [14]:
sample

french  \
0                 Et ne viens surtout pas à l'école !   
1                             Ce mouton-là en entier.   
2                              C'est quand tu es venu   
3     Si tu veux, retourne au centre de Saint-Louis !   
4   Je veux que tu viennes le matin à cinq heures ...   
5        Ce matin ou ce soir c'est moi qui suis ici !   
6                               J'ai aperçu un homme.   
7                          Là où il y a de la lumière   
8                                   J'ai vu la femme.   
9                        Aucun mouton ne s'est égaré.   
10            Les gens n'eussent pas été des maîtres.   
11                       J'ai vu des vaches d'Europe.   
12  Les enfants ont cherché à l'intérieur, à l'ext...   
13                                  Vous fûtes Laobé.   
14         L'homme a tenu aux enfants de bons propos.   
15                        Il est venu à cause de toi.   
16                   Je connaissais cet individu-là !   
17                         D'ailleurs, je m'en vais !   
18                       J'ai aperçu une jeune fille.   
19   Il n'est de Ardo du Dieri que tu ne connaisses !   
20                                  Que pleure-t-il ?   
21                                Honore tout homme !   
22        L'enfant dit d'entrer pour qu'il vous voie.   
23                       C'est cette ville de Sérère.   

                                             wolof  
0                         Te bul ñëw jaŋgu ba de !  
1                                Xar moomule mépp.  
2                                  Bi ŋga ñëwée la  
3                   Dellu, soo bëggée, biir Ndar !  
4    Bëgg naa ci juróomi waxtu, ci subë, ŋga ñëw !  
5         Maa fiy nekk tày ci ŋgoon, tày ci suba !  
6                                 Séen naa aw nit.  
7                                 Ci biir fiy leer  
8                               Gis naa jigéen ju.  
9                                 Menn xar réerul.  
10                     Nit ñi doonkoonuñu ay kaaŋ.  
11                             Gis naa nagu Tugël.  
12  Xale yi set nañu ci biir, ci biti gisuñu dara.  
13                               Laobe ŋgeen woon.  
14            Góor gi wax na xale yi wax ju rafet.  
15                               Dikk na ndax yaw.  
16                         Xamoon naa waa joojee !  
17                           Te sax, ma ŋgii dem !  
18                               Séen naa as ndaw.  
19               Amul benn ardow Jéeri boo xamul !  
20                                   Lu muy jooy ?  
21                              Teralal bépp nit !  
22              Xale bi ne ŋgeen dugg mu gis leen.  
23                           Gii dëkku Séeréer la.

It only kept 26 samples. Let us make the same for all the sentences.

In [15]:
diagne_sentences = select_by_length(diagne_sentences, min_len=3)

corpora_v4 = select_by_length(corpora_v4, min_len=3)

Let us verify how many sentences we have kept.

In [16]:
# for diagne's sentences
diagne_sentences.shape[0]

1329

In [17]:
# in corpora v4
corpora_v4.shape[0]

2213

We removed $651$ sentences from the corpora v4 and $645$ sentences from the diagne's sentences.

Let us save the results to make again hyperparameter search with them.

In [18]:
# save the new version of the diagne's sentences in red_sentences.csv
diagne_sentences.to_csv('data/extractions/new_data/red_sentences.csv', index = False)

# save the new version of the corpora
corpora_v4.to_csv('data/extractions/new_data/corpora_v5.csv', index = False)

### Recuperate additional sentences 

In [19]:
corpora_v4 = pd.read_csv('data/extractions/new_data/corpora_v4.csv')

diagne_sentences = pd.read_csv('data/extractions/new_data/sentences.csv')

Let us recuperate additional sentences to augment the corpora.

- First file:

In [20]:
path = 'data/additional_documents/omniglot/sentences_french_wolof.txt'

with open(path, 'r', encoding='utf-8') as f:
    
    # recuperate lines
    omn_1 = f.readlines()
    
# recuperate sentences
omn_1_ = {
    'french': [],
    'wolof': []
}

for omn in omn_1:
    
    omn = omn.split('\t')
    
    sentences = [s.strip() for s in omn if s.strip() != '']
    
    omn_1_['french'].append(sentences[0])
    
    omn_1_['wolof'].append(sentences[1])

In [21]:
pd.options.display.max_rows = 74
omn_1_ = pd.DataFrame(omn_1_).head(74)

# print the first sentences from omniglot
omn_1_.head(74)

french  \
0                                    Bienvenue   
1                                        Salut   
2                                        Salut   
3                                        Salut   
4                             Comment vas-tu ?   
5                              Es-tu en paix ?   
6                             Comment vas-tu ?   
7                     Paix seulement, et toi ?   
8      Je suis seulement toi, comment vas-tu ?   
9      Ça fait longtemps que je ne t'ai pas vu   
10                     Comment t'appelles-tu ?   
11                     Comment t'appelles-tu ?   
12                     Comment t'appelles-tu ?   
13                             Mon nom est ...   
14                             Mon nom est ...   
15                           Je m'appelles ...   
16                             D'où viens-tu ?   
17                           D'où venez-vous ?   
18                             Je viens de ...   
19                             Je viens de ...   
20                    Plaisir de te rencontrer   
21                    Content de te rencontrer   
22            Je suis content de te rencontrer   
23                                   Bon matin   
24                            Bonne après-midi   
25                                Bonne soirée   
26                                   Au revoir   
27                                   Au revoir   
28                              Bonne chance !   
29                               Bonne santé !   
30                      Aies une bonne journée   
31                                  Bon voyage   
32                              Comprends-tu ?   
33                                Je comprends   
34                         Je ne comprends pas   
35                        Je ne le connais pas   
36                           Je ne le sais pas   
37      S'il te plaît, parles plus lentement !   
38                     Répètes s'il te plaît !   
39                       Ecris s'il te plaît !   
40                         Parles-tu anglais ?   
41                           Parles-tu Wolof ?   
42                                 Oui, un peu   
43            Comment dis t-on ceci en Wolof ?   
44                                Excuse-moi !   
45                          Combien ça coute ?   
46                                Pardonne-moi   
47                                  Si tu veux   
48                                       Merci   
49                                       Viens   
50                               Venez en paix   
51                     Où sont les toilettes ?   
52  Ce garçon peut te payer pour n'import quoi   
53             Cette jeune fille va tout payer   
54             Accepte-tu de danser avec moi ?   
55             Accepte-tu de danser avec moi ?   
56                               Tu me manques   
57                                  Je t'aimes   
58                                Je t'admires   
59                                  Je t'aimes   
60            Je te souhaites de récupérer tôt   
61                                  Vas t'en !   
62                                  Vas t'en !   
63                    Laisses moi tranquille !   
64                    Laisses moi tranquille !   
65                                  À l'aide !   
66                                       Feu !   
67                                    Arrête !   
68                         Appelle la police !   
69                           Mes voeux de noël   
70                      Mes voeux de nouvel an   
71                         Mes voeux de paques   
72                        Mes voeux de tabaski   
73                    Mes voeux d'anniversaire   

                                          wolof  
0                                        Merhbe  
1                                    Na nga def  
2                                  Na ngeen def  
3                                Salaam aleekum  
4                                 Jaam 

- Second file

In [22]:
path = 'data/additional_documents/omniglot/sentences_french_wolof_2.txt'

with open(path, 'r', encoding='utf-8') as f:
    
    # recuperate lines
    omn_2 = f.readlines()
    
# recuperate sentences
omn_2_ = {
    'french': [],
    'wolof': []
}

for omn in omn_2:
    
    omn = omn.split('–')
    
    sentences = [s.strip() for s in omn if s.strip() != '']
    
    omn_2_['wolof'].append(sentences[0])
    
    omn_2_['french'].append(sentences[1])

In [23]:
pd.options.display.max_rows = 74
omn_2_ = pd.DataFrame(omn_2_).head(74)

# print the first sentences from omniglot
omn_2_.head(74)

french                    wolof
0          Comment allez-vous ?           Na ngeen def ?
1                Bonne matinée.         Jaam nga fanane.
2                   Bonne nuit.           Fanaanal jaam.
3             À une autre fois.               Ba beneen.
4                 Je t'en prie.                   Agsil.
5              Je vous en prie.        Agsileen ak jaam.
6                       Pardon.                 Baal ma.
7                          Oui.                     Wau.
8                          Non.                 Deh-det.
9               Es-tu en paix ?            Jaam nga am ?
10             Comment vas-tu ?            Jaam nga am ?
11              Paix seulement.                Jaam rek.
12             Je suis en paix.                Jaam rek.
13                     Et toi ?                Yow nag ?
14      C'est comment ton nom ?          Naka-nga sant ?
15            D'où venez-vous ?        Fan ngeen joghe ?
16            Je viens des USA.   Maa ngi joghe les USA.
17           Parles-tu Anglais?         Deg nga Angale ?
18         Parles-tu Français ?        Deg nga Faranse ?
19  Je parle seulement anglais.    Angale rekk laa degg.
20     Je parle un peu anglais.  Degg naa tuuti Faranse.
21      Je ne parles pas Wolof.     Mahn deggumah Wolof.
22   Je ne parles pas Français.   Mahn deggumah Faranse.
23            Je n'entends pas.                 Degguma.
24               J'aimerais … .        Dama bahggoon … .
25                   Où est … ?              Fahn la … ?
26                Est-ce loin ?               Soreh na ?
27            C'est devant toi.               Cha kanam.
28                  Devant toi.               Cha kanam.
29                      Gauche.                Chammoon.
30                       Droit.                Ndeyjoor.
31                 Mais entre !           Dugghal waay !
32            Combine ça vaut ?               Lii naata?
33             C'est trop cher.           Seer na torob.

- Third file

In [24]:
path = 'data/additional_documents/omniglot/sentences_french_wolof_3.txt'

with open(path, 'r', encoding='utf-8') as f:
    
    # recuperate lines
    omn_3 = f.readlines()
    
# recuperate sentences
omn_3_ = {
    'french': [],
    'wolof': []
}

for omn in omn_3:
    
    omn = omn.split('\t')
    
    sentences = [s.strip() for s in omn if s.strip() != '']
    
    omn_3_['wolof'].append(sentences[0])
    
    omn_3_['french'].append(sentences[1])
      
        

In [25]:
pd.options.display.max_rows = 74
omn_3_ = pd.DataFrame(omn_3_).head(74)

# print the first sentences from omniglot
omn_3_.head(74)

french                             wolof
0             Que la paix soit avec toi.                 Salaam maaleekum.
1       Que la paix soit avec toi aussi.                  Maaleekum salaam
2                        Es-tu en paix ?                     Jaama ngaam ?
3                        Paix seulement.                        Jamma rek.
4                Comment va ta famille ?                  Ana waa kër gi ?
5    Où sont les membres de ta famille ?                  Ana waa kër gi ?
6                       Ils sont là-bas.                         Ñunga fa.
7             J'espère qu'ils vont bien.                 Mbaa defuñu dara.
8               Rien ne va mal avec eux.                 Mbaa defuñu dara.
9                    Non, ils vont bien.             Déedéet, defuñu dara.
10              Rien ne va mal chez-eux.             Déedéet, defuñu dara.
11         Comment se passe le travail ?                  Naka ligéey bi ?
12             J'y suis avec, pas à pas.  Maangi ci kowam, ndanka, ndanka.
13         J'y suis avec, petit à petit.  Maangi ci kowam, ndanka, ndanka.
14                      Comment vas-tu ?                       Nanga def ?
15                      Comment vas-tu ?                         Noo def ?
16  Comment vont les gens de la maison ?                 Naka waa kër ga ?
17               Comment va la famille ?                 Naka njaboot gi ?
18                         Je vais bien.                    Maangi fi rek.
19        Qu'est-ce que tu as rapporté ?                     Lan nga indi?
20          J'ai rapporté des nouvelles.                  Xibaar laa indi.
21               Je suis dans la maison.                 Maangi ci nég bi.
22               Ils se tiennent dehors.              Ñungi taxaw ci biti.
23         As-tu passé la nuit en paix ?                Jaama nga fanaan ?
24         Comment se passe la matinée ?                   Naka subba si ?
25             La matinée se passe bien.               Subba sangi fi rek.
26      As-tu passé la matinée en paix ?                   Jaama nga endu?
27       Comment se passe l'après-midi ?                   Naka bëcëk bi ?
28           L'après-midi se passe bien.              Bëcëk baangi fi rek.
29         Comment se passes la soirée ?                   Naka ngoon si ?
30              La soirée se passe bien.              Ngoon saangi fi rek.
31            Comment se passe la nuit ?                   Naka guddi gi ?
32                La nuit se passe bien.               Guddi gaangi fi rek
33                          Comment ...?                         Noo ... ?

- Let us retrieve sentences, that was created by internet users in our following platform [addition_sentences_platform](https://french-wolof-translation.streamlit.app/), from our MongoDB Atlas Database.

In [26]:
# initialize the path to the csv files
sentences_path = 'data/additional_documents/hugging_face/wolof_french.csv'
deleted_path = 'data/additional_documents/hugging_face/deleted.csv'

# initialize the server uri and database name
uri = "mongodb+srv://oumar199:Jacksparrow360@woloftranslationcluster.u0gk7.mongodb.net/?retryWrites=true&w=majority"
# uri = "mongodb+srv://oumarkanetest:Jacksparrow360__@woloftranslationcluster.u0gk7.mongodb.net/?retryWrites=true&w=majority"
database_name = 'WolofTranslation'

# initialize the database manager 
database_manager = TranslationMongoDBManager(uri, database_name)


# save locally the data sets
database_manager.save_data_frames(sentences_path, deleted_path)

hug_sents = pd.read_csv(sentences_path)

In [27]:
hug_sents.head()

french  \
0                   J'arrive tout de suite chez toi.   
1  J'en suis sûr, cette photo ci c'est la photo p...   
2  Je vois devant moi une photo sur laquelle beau...   
3  Ceux-ci sont des personnes qui sont sortis pou...   
4  Je pense que la photo que je vois là, c'est un...   

                                               wolof  
0                       Léegui léegui ma egg sa kër.  
1  Waaw nataal bii nataal la boob ay nit ñu baree...  
2  Nataal bii maa ngi ciy janloog haa ay nit yu b...  
3  Lii, ay nit lañu yu génn di ñaxtu. Jëm yi nag ...  
4  Man de nataal bii ma gis nii, nataal la boo xa...

-------------------

Nous devons ajouter des espaces autour des expressions entre apostrophes.

In [28]:
# # rectifier les phrases avec la fonction suivante
# def rectify_apos(sentence: str): # with the help of chat gpt
    
#     return re.sub(r"(?<![\wàâçéèêëîïôûùüÿñæœ])'([^']*?)'(?![\wàâçéèêëîïôûùüÿñæœ])", r" '\1' ", sentence)



In [29]:
# # add modification to all sentences
# hug_sents['french'] = hug_sents['french'].apply(rectify_apos)
# hug_sents['wolof'] = hug_sents['wolof'].apply(rectify_apos)

In [30]:
# hug_sents.to_csv('verification.csv', index=False)

In [31]:
# # recuperate the sentences from the csv file
# hug_sents = pd.read_csv('verification.csv')

In [32]:
# # add the modification to the mongo database sentences according to their ids
# def add_modification(sentences: pd.DataFrame, database_manager: TranslationMongoDBManager):
    
#     for i in list(sentences.index):
        
#         # recuperate the sentences
#         french_sent = sentences.loc[i, 'french']
        
#         wolof_sent = sentences.loc[i, 'wolof']
        
#         # add the modification
#         database_manager.update_document(i, 
#                                          {
#                                                 'french': french_sent,
#                                                 'wolof': wolof_sent
#                                          })

# add_modification(hug_sents, database_manager)

------------

In [33]:
# # print some documents inside the sentences collection using the uri
# def write_documents(uri):
        
#     from pymongo import MongoClient
    
#     # connect to the database
#     client = MongoClient(uri)
    
#     # get the database
#     db = client['WolofTranslation']
    
#     # get the sentences collection
#     collection = db['sentences']
    
#     # recuperate some documents
#     documents = collection.find()
    
#     # write the documents in a file
#     for doc in documents:
        
#         print(str(doc))
#         print(len(documents))
            
#     # close the connection
#     client.close()
    
# write_documents(uri)


In [34]:
pd.options.display.max_rows = 74

hug_sents.head(74)

french  \
0                    J'arrive tout de suite chez toi.   
1   J'en suis sûr, cette photo ci c'est la photo p...   
2   Je vois devant moi une photo sur laquelle beau...   
3   Ceux-ci sont des personnes qui sont sortis pou...   
4   Je pense que la photo que je vois là, c'est un...   
5   Bonjour ! Ceci est une photo. Sur cette photo ...   
6   Ceci est une photo sur laquelle je vois un poi...   
7   Ceci est ce qu'on appelle une douche, qu'on a ...   
8   Sur cette photo il semble qu'il y est un table...   
9   Sur cette photo, c'est une personne de couleur...   
10  Jusqu'à présent, on voit sur la photo des sold...   
11  Ceci, cependant, est un bol. Il y a du sucre d...   
12  Sur cette photo, il y a d'une salle dans la qu...   
13  Ceci est un ancien milieu avec une constructio...   
14  Ceux-ci sont de vieilles cases à l'intérieur d...   
15  Sur cette photo ci, j'y aperçois du beurre ; i...   
16  Ceci est une bouche qu'on a conçue, dans laque...   
17  Ici, cependant, il semble qu'il y est une port...   
18  Sur ma photo qu'on m'a envoyée, il y a apparem...   
19  Ceux-là, cependant, sont des cases. Celle qui ...   
20  Sur cette photo se trouvent deux présidents. I...   
21  Oui encore, je vois que ceci est une automobil...   
22  Sur cette photo, il y a un homme et une femme,...   
23  Ceci est une tasse et une assiette. On y verse...   
24  Ceci est un lieu d'habitation qui contient un ...   
25  Oui, sur cette photo, c'est des portes en gris...   
26  Cela est une photo. Je vois, dessus, deux femm...   
27  Cela est ce qu'on appelle  'La tour la plus ha...   
28  Ceux-là, aussi, sont des gendarmes. Ils siègen...   
29  Ceux-là sont des femmes policières. Elles sont...   
30  Sur cette photo ci, cependant, il y a des pers...   
31  Oui le parent, je vois que ceci c'est une théi...   
32  Sur cette photo ci il y a une assurément une c...   
33  Sur la photo que vous m'avez envoyée j'y ai vu...   
34  Sur cette photo j'y ai vu une case qui se trou...   
35  Sur cette photo ci, on dirait des autochtones ...   
36  Sur cette photo ci cependant, il y a un panier...   
37  Sur cette photo, j'y vois quelque chose qui re...   
38  Sur la photo que vous m'avez envoyée, j'y ai v...   
39  Ici j'y ai vu deux assiettes blanches.Parmi le...   
40  Celui là, j'ai vu que c'est un garçon, debout ...   
41  Sur cette photo ci, j'y ai vu des chambres fai...   
42  Oui ça, il semble que ce soit une fleur, mais ...   
43  À l'intérieur de cette photo je vois beaucoup ...   
44  Je vois une large chambre avec un lit pour pas...   
45  Ceci ressemble à un bahut. Un bahut chocolat d...   
46  Je sens que ceux-là sont, vraisemblablement, d...   
47  C'est des enfants qui jouent avec leurs jouets...   
48  C'est l'entrée de Louga. C'est à la frontière ...   
49  Ceux-là sont des garçons montés sur des chevau...   
50  Celui-là que tu vois c'est le dirigeant du pay...   
51  Cette photo ci est une photo sur laquelle il y...   
52  Sur cette photo, il y a une chose qui ressembl...   
53  Sur cette photo ci que je vois il y a des pers...   
54  Sur la photo que j'ai vue, j'y ai vu une jolie...   
55  Sur cette photo on voit ce qu'on appelle  'poi...   
56  Sur la photo que vous m'avez envoyée, j'ai vu,...   
57  Ceci est un four. C'est ce qu'on appelle une c...   
58  Sur la photo qu'on m'a envoyée, j'y ai vu un c...   
59  Sur cette photo, je suis face à une chaise pei...   
60  Ceci est un long bâtiment blanc. Il contient b...   
61  C'est des poissons, oui. Ils sont de couleur b...   
62  Ah sur cette photo ci cependant, il y a un poi...   
63  Ceci est un long et jolie bâtiment. On a vu de...   
64  Ceci se trouve dans les toilettes. <mask> cett...   
65  Ceux-ci sont des portes que les menuisiers mét...   
66  Sur la photo que vous m'avez envoyée, j'y ai v...   
67  Cela est une chaise de la rue  'la jardin publ...   
68  Ça c'est nos parents Laobés qui l'ont conçu. J...   
69  Ceci est un lieu d'habitation très charmant, t...   


Let us add them into the diagne's sentences and the version 4 of the corpora and remove duplicated rows.

- On corpora version 4

In [35]:
corpora_v6 = pd.concat((corpora_v4, omn_1_, omn_2_, omn_3_, hug_sents), axis=0)

# drop duplicated rows
print(sum(corpora_v4.duplicated()))

9


In [36]:
corpora_v6.drop_duplicates(inplace=True)

9 rows were deleted.

- On corpora version 5

In [37]:
ad_sentences = pd.concat((diagne_sentences, omn_1_, omn_2_, omn_3_, hug_sents), axis=0)

# drop duplicated rows
print(sum(ad_sentences.duplicated()))

6


In [38]:
ad_sentences.drop_duplicates(inplace=True)

6 rows were deleted.

### Web Scraping

This step involve to scrape some websites to obtain more sentences using `selenium` and `BeautifulSoup`. It will require to meticulously investigate each website and identify the elements to trigger and to extract. We will also need to identify the transformation to add to the different sentences. All the transformations, excepted the sentences' capitalization, are already implemented and can be directly used. 

We will scrape the following websites: 

- url1: [corporan.huma-num.fr](https://corporan.huma-num.fr/Lexiques/indexM.php?controle=afficheDico&action=main&ficXML=DicoWolof.xml&langue=Wolof)

In [54]:
url1 = "https://corporan.huma-num.fr/Lexiques/indexM.php?controle=afficheDico&action=main&ficXML=DicoWolof.xml&langue=Wolof"

#### Corporan.huma-num scraping

Let us add the necessary libraries.

In [51]:
from wolof_translate.utils.extract_new_sentences import NewSentenceExtraction
from wolof_translate.utils.sent_transformers import TransformerSequences
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.ui import WebDriverWait
from wolof_translate.utils.sent_corrections import *
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.chrome import service
from selenium.webdriver.common.by import By
from urllib import request as rq
from selenium import webdriver
from bs4 import BeautifulSoup
from tqdm import tqdm
from typing import *
import pandas as pd

We need to identify the element containing the sentences. It is easy in that case since they are identified by their classes:

- The french sentences are identified by the `xe` class
- The wolof sentences by the `xv` class.

Since the website consists of a dictionary of word, we must click on one of them to display the sentences. The words' elements are not links but it use a ajax function. Then we must use trigger event to display the different pages. The words are represented by the `div` tag and each of them has a unique id which is simply the number of the word. We have `9965` words so the ids range from 1 to 9965.

Let us begin by initializing the drive. We will use the chrome driver in our case.

In [2]:
# initialize the option
options = webdriver.ChromeOptions()

# we will add the headless option which will not render the browser
options.add_argument('--headless')

# let us recuperate the service
s = service.Service('C:\Program Files\Chromedriver\chromedriver.exe')

# let us recuperate the driver
driver = webdriver.Chrome(service=s, options=options)



Let us request to go the website.

In [57]:
# send a get request to the website
driver.get(url1)

Let us initialize the classes.

In [52]:
# initialize the class map
class_map = {
    'xe': 'french',
    'xv': 'wolof'
}


Let us initialize the transformer.

In [53]:
transformer = TransformerSequences(remove_mark_space, delete_guillemet_space, add_mark_space)

Let us create a function which get recuperate the content and with beautiful soup recuperate the sentences. We will use an implicit wait at each time since we don't know what element to wait for: Not all pages contain sentences. We will wait 0.2 seconds before getting the sentences.

In [60]:
def get_sentences_from_page(driver: webdriver.chrome.webdriver.WebDriver, sentences: Dict[str, list], transformer: Union[TransformerSequences, None] = None, replaces: dict = {'…': '...'}, end_marks = ['.', '!', '?', '...']):
    
    # let us recuperate the content of the current page
    content = driver.page_source
    
    # recuperate the document object
    doc = BeautifulSoup(content, 'lxml')
    
    # recuperate the elements
    elements = {class__: doc.find_all('span', class_ = class__) for class__ in sentences}
    
    if elements != {key: [] for key in sentences}:
        
        # count the number of sentences that we got
        sent_count = len(elements[list(sentences)[0]])
        
        # for each group of sentences we will add some transformations
        for s in range(sent_count):
            
            # initialize the common end mark
            end_mark = None
            
            # recuperate the sentences and add to the dictionary of sentences
            for class__ in sentences:
                
                # recuperate the current sentence
                sentence = elements[class__][s].text.strip()
                
                # replace marks
                for replace in replaces:
                    
                    sentence = sentence.replace(replace, replaces[replace])
                
                if not sentence[-1] in end_marks:
                    
                    # add an end mark if the end mark is got from one of the previous sentences
                    if not end_mark is None:
                        
                        sentence += end_mark
                    
                    else: # add a period mark
                        
                        sentence += '.'
                
                else: # if an end mark is identified recuperated it as the end mark of the other sentences
                    
                    end_mark = sentence[-1]
                
                # capitalize the current sentence
                sentence = sentence[0].upper() + sentence[1:]
                
                # add transformations to the sentence
                if transformer: sentence = transformer(sentence)[0]
                
                # add the sentence to the dictionary of sentences
                sentences[class__].append(sentence)
            
    # return the sentences
    return sentences
    

In [ ]:
# calculate the approximate number of minutes
0.5*9956/60

82.96666666666667

Initialize the sentences 🛑.

In [54]:
# recuperate the saved sentences
scraped_sentences = pd.read_csv('data/extractions/new_data/scraped_sents.csv').to_dict('list')

# initialize the sentences
sentences = {key: scraped_sentences[class_map[key]] for key in class_map}

# current_id
current_id = 9836

# initialize the ids
word_ids = list(range(current_id, 9966))

Let us recuperate the sentences by clicking recursively at each word.

In [ ]:
# initialize the time sleep previously introduced
time_sleep = 0.6

# initialize a save step
save_step = 20

# recuperate the sentences of each word
for id in tqdm(word_ids):
        
    # click next to the word
    word = driver.find_element(By.ID, id)
    driver.execute_script('arguments[0].click();', word)
    
    # add a time sleep
    driver.implicitly_wait(time_sleep)
    
    # recuperate the sentences
    sentences = get_sentences_from_page(driver, sentences, transformer, replaces={'…': '...', '\xa0': ' ', 'qqn': "quelqu'un"})
    
    # save the sentences
    if id % save_step == 0:
        
        pd.DataFrame({value: sentences[key] for key, value in class_map.items()}).to_csv('data/extractions/new_data/scraped_sents.csv', index = False)
    
    

100%|██████████| 130/130 [03:41<00:00,  1.70s/it]


Not all sentences were recuperated. Let us make it again to recuperate additional sentences from the website.

Initialize the sentences 🛑.

In [68]:
# recuperate the saved sentences
scraped_sentences_2 = pd.read_csv('data/extractions/new_data/scraped_sents2.csv').to_dict('list')

# initialize the sentences
sentences_2 = {key: scraped_sentences_2[class_map[key]] for key in class_map}

# current_id
current_id = 300

# initialize the ids
word_ids = list(range(current_id, 9966))

Let us recuperate the sentences by clicking recursively at each word.

In [66]:
# initialize the time sleep previously introduced
time_sleep = 0.6

# initialize a save step
save_step = 20

# recuperate the sentences of each word
for id in tqdm(word_ids):
        
    # click next to the word
    word = driver.find_element(By.ID, id)
    driver.execute_script('arguments[0].click();', word)
    
    # add a time sleep
    driver.implicitly_wait(time_sleep)
    
    # recuperate the sentences
    sentences_2 = get_sentences_from_page(driver, sentences_2, transformer, replaces={'…': '...', '\xa0': ' ', 'qqn': "quelqu'un"})
    
    # save the sentences
    if id % save_step == 0:
        
        pd.DataFrame({value: sentences_2[key] for key, value in class_map.items()}).to_csv('data/extractions/new_data/scraped_sents2.csv', index = False)
    
    

100%|██████████| 9666/9666 [5:19:39<00:00,  1.98s/it]   


Let us take concatenate the two data frames and drop duplicated rows.

In [70]:
# concatenate the translations
scraped_sentences = pd.concat((pd.DataFrame(scraped_sentences), pd.DataFrame(scraped_sentences_2)))

In [72]:
# drop duplicated translations
scraped_sentences.drop_duplicates(inplace=True)

Let us verify how many translations we have.

In [74]:
# print the number of new translations
scraped_sentences.shape[0]

9388

Let us save the result.

In [76]:
scraped_sentences.to_csv('data/extractions/new_data/scraped_sents.csv', index=False)

We obtained `9388` translations from the first scraped website. Let us now make pre-processing on it. We will based on the following strings to make it:
- The parentheses containing 'ou...' 
- The slashes '/'
- The parentheses: We mostly want to delete them all
- Replace 'qqch' by 'quelque chose'
- Replace ';.' by '.'
- Replace 'œ' by 'oe'
- Identify the sentence with only ')' but not '('.

Let us create a function which can recuperate the sentences with regular expressions and another function which can replace an expression by another.

In [45]:
# %%writefile wolof-translate/wolof_translate/utils/re_extract.py
from typing import *
import re

def extract_with_re(translations: pd.DataFrame, columns: list = [], regex: Union[str, None] = None, recup_csv: str = 're_extract.csv'):
    
    # get all columns if no column is given
    if not columns: columns = list(translations.columns)
    
    for column in columns:
        
        assert column in set(translations.columns)
    
    # if not regular expression is given then return in the csv file all the sentences
    if not regex is None:
        
        # initialize the dictionary of the concerned sentences
        re_sents = {
            column: [] for column in columns
        }
        
        # at each index we will verify if a column respect the regular expression
        for i in range(translations.shape[0]):
            
            # make verification for each column
            re_true = False
            
            for column in columns:
                
                if re.match(regex, translations.loc[i, column]):
                    
                    re_true = True
                
            # add the line inside the set of translations to investigate
            if re_true:
                
                for column in columns:
                        
                        re_sents[column].append(translations.loc[i, column])
                
                # remove the sentences from the DataFrame
                translations.drop(index=[i], inplace=True)
            
        # save the resulted sentences in order to modify (it will wait until you finish the modifications)
        re_sents = pd.DataFrame(re_sents)
        
        print("You recuperated the following sentences:")
        print(re_sents)
        
        re_sents.to_csv(recup_csv, index=False)
        
        alert_ = ''
        
        while True:
            
            finish = input(f"{alert_}Provide Yes(y) if you finish and No(n) if not yet!")
            
            if finish == 'y':
                
                break
            
            elif finish != 'n':
                
                alert_ = "You cannot provide a command which is different from y (for Yes) and n (No)"
            
        # recuperate the modifications
        re_sents_2 = pd.read_csv(recup_csv)
        
        # add the sentences inside the original data frame and return the results
        translations = pd.concat((re_sents_2, translations))
        
        translations.index = range(translations.shape[0])
        
        return translations, re_sents_2
    
    return None
    
def replace_expression(translations: pd.DataFrame, columns: list = [], regex: Union[str, None] = None, new_expression: str = ''):
    
    # get all columns if no column is given
    if not columns: columns = list(translations.columns)
    
    for column in columns:
        
        assert column in set(translations.columns)
    
    # if not regular expression is given then return in the csv file all the sentences
    if not regex is None:
        
        # initialize the dictionary of the concerned sentences
        re_sents = {
            column: [] for column in columns
        }
        
        # at each index we will verify if a column respect the regular expression before making replacement
        for i in range(translations.shape[0]):
            
            for column in columns:
                
                expressions = re.findall(regex, translations.loc[i, column])
                
                if expressions != []:
                
                    translations.loc[i, column] = re.sub(regex, new_expression, translations.loc[i, column]).strip()

                    for column in columns:
                        
                        re_sents[column].append(translations.loc[i, column])
            
        # save the resulted sentences in order to modify (it will wait until you finish the modifications)
        re_sents = pd.DataFrame(re_sents)
        
        print("You recuperated the following sentences:")
        print(re_sents)
    
        return translations, re_sents
    
    return None

- Let us recuperate all translations containing parentheses with 'ou' and verify which of them will need to separated as multiple translations.

In [21]:
scraped_sentences, re_sents = extract_with_re(pd.DataFrame(scraped_sentences), regex='.*\(ou.*\).*')

You recuperated the following sentences:
                                               french                                      wolof
0   Je m'occupe de toi; je suis à toi. (ou encore)...                            Maa ngi ci yow.
1               C'est bien (ou bien) d'accord; ça va.                                   Baax na.
2   Il est bel et bien au courant (ou bien) il l'a...                    Yég na ko ba bëgga dee.
3                C'est pareil (ou bien) c'est unique.                                   Benn la.
4    Ils sont semblables (ou bien) ils ne font qu'un.                                 Benn lañu.
..                                                ...                                        ...
27         Être soupe au lait (ou bien) être frustré.                                  Tàng xol.
28                                       De bon cœur.  Ci xol bu sedd (ou bien) ci xol bu tàlli.
29       Vous êtes égaux (ou bien) vous êtes pareils.                                 

- Let us recuperate all translations containing slashes.

In [34]:
scraped_sentences, re_sents = extract_with_re(scraped_sentences, regex='.*(/).*')


You recuperated the following sentences:
                                              french                                              wolof
0  Qu'est-ce qui te prend ? / qu'est-ce qui t'arr...                                 (loc.) Lu la dal ?
1  Fais des boulettes plus grosses / fais davanta...                                    Yokkal dank yi.
2    Oh la, Omar ! / Arrête, Omar ! (Tu as fini ?) .                                         Eey, Omar.
3  J'en ai plus que toi / j'en ai un plus grand (...                                      Maa la ëpple.
4                Le voilà ! / Voilà / nous-y voilà !                                  (Lov.)Mu ngoogu !
5  Celui qui meurt au marché fait lui-même son an...  Ku dee marse, yaa tàgge sa bopp // ku dee ca j...
6  Il n'a pas mangé / il n'est pas en train de ma...                                            Lekkul.
7  Je n'ai pas mangé / je ne suis pas en train de...                                 Lekkuma lekkul ma.


- Separate in multiple ones the translations with parentheses

In [37]:
scraped_sentences, re_sents = extract_with_re(scraped_sentences, regex='.*\(.*\).*')

You recuperated the following sentences:
                                                french                                              wolof
0        Ces touffes (de cheveux), c'est leur coutume.                            Jubb yii, seen aada la.
1    Avant de tirer sur une bête, je vois d'abord s...  Bala may fital aw rab, dinaa xool ndax mu ngi ...
2       L'interdiction est temporaire (ne durera pas).                                   Aaye bi du yàgg.
3    Il prête sa charrette (à qui veut) uniquement ...  Li muy abalaate saretam bi yépp dey, bëgga fal...
4                      Les enfants sont arrivés (ici).                                 Xale yi agsi nañu.
..                                                 ...                                                ...
167  J'ai appelé ce village Njar-Meew (lait coupé) ...  Lu naqari la mu ma doon def, di ma ko yéjje, m...
168  Il dit qu'on a libéré les invités (ils peuvent...                           Mu ne yewwi nañu gan yi.
169  

- Delete parentheses

In [38]:
scraped_sentences, re_sents = replace_expression(scraped_sentences, regex='\(.*\)')

You recuperated the following sentences:
                                                french                                              wolof
0                    Ces touffes , c'est leur coutume.                            Jubb yii, seen aada la.
1    Avant de tirer sur une bête, je vois d'abord s...  Bala may fital aw rab, dinaa xool ndax mu ngi ...
2                      L'interdiction est temporaire .                                   Aaye bi du yàgg.
3     Il prête sa charrette  uniquement pour être élu.  Li muy abalaate saretam bi yépp dey, bëgga fal...
4                           Les enfants sont arrivés .                                 Xale yi agsi nañu.
..                                                 ...                                                ...
168                 J'ai appelé ce village Njar-Meew .  Lu naqari la mu ma doon def, di ma ko yéjje, m...
169                Il dit qu'on a libéré les invités .                           Mu ne yewwi nañu gan yi.
170  

- Replace 'qqch' by 'quelque chose'

In [47]:
scraped_sentences, re_sents = replace_expression(scraped_sentences, regex='qqch', new_expression='quelque chose')


You recuperated the following sentences:
                                               french                                              wolof
0   Il y a quelque chose  dans mes habits; tiens l...  Am na luy dox ci sama yére yi; téyeel ma xale ...
1   Dis-moi quelque chose qui puisse apporter la p...         Wax ma lu mëna amal jàmm ci sunu réew mi !
2   Je pensais qu'il m'appelait pour quelque chose...   Dama defe woon ne lu am-maanaa la ma doon wooye.
3   Le réparateur est arrivé. Qui aurait quelque c...        Baay-jagal dikk na. Ku amoon looy defarlu ?
4   Ses propos ont été brefs; lui aurais-tu fait q...       Waxam ji dafa bar; xanaa danga ko def dara ?
..                                                ...                                                ...
42  Quand on fait quelque chose, il est normal d'e...                            Kuy xalam, di ci jaayu.
43  Quand on reçoit quelque chose d'une personne q...  Bu la ku xér di may, sa xol sedd; waaye bu nge...
44  Ce dont on

- Replace ';.' by '.'

In [49]:
scraped_sentences, re_sents = replace_expression(scraped_sentences, regex=';\.', new_expression='.')

You recuperated the following sentences:
                                                french                                              wolof
0    Il y a quelque chose  dans mes habits.tiens l'...  Am na luy dox ci sama yére yi; téyeel ma xale ...
1    Il y a quelque chose  dans mes habits.tiens l'...  Am na luy dox ci sama yére yi.téyeel ma xale b...
2    Ses propos ont été brefs.lui aurais-tu fait qu...       Waxam ji dafa bar; xanaa danga ko def dara ?
3    Ses propos ont été brefs.lui aurais-tu fait qu...        Waxam ji dafa bar.xanaa danga ko def dara ?
4    Ça, il faudra tout faire pour l'avoir, c'est q...  Lii dey, nanga ci gar sa bakkan.lu lay amal nj...
..                                                 ...                                                ...
443           Il est fâché.il ne veut même pas manger.                       Dafa aru; nanguwul sax lekk.
444           Il est fâché.il ne veut même pas manger.                        Dafa aru.nanguwul sax lekk.
445  

In [53]:
# function created to add correction 
# def correction(text: str):
    
#     charac = ''
    
#     for i in range(len(text[:-1])):
        
#         letter = text[i]
        
#         if letter == '.' and re.match('[a-zA-Z]', text[i+1]):
            
#             charac = charac + '; '
        
#         else:
            
#             charac = charac + letter
        
#     return charac + text[-1]

- Replace 'œ' by 'oe'

In [58]:
scraped_sentences, re_sents = replace_expression(scraped_sentences, regex='œ', new_expression='oe')


You recuperated the following sentences:
                                               french                                       wolof
0   Une femme de moeurs légères ne se marie pas fa...                 Jigéen juy fo du yomba séy.
1                            Elle est ma jeune soeur.                               Sama rakk la.
2                                       De bon coeur.                            Ci xol bu tàlli.
3                                       De bon coeur.                             Ci xol bu sedd.
4                      Chante, nous ferons le choeur.                           Woyal, nu awu la.
..                                                ...                                         ...
93              Ton souhait ne te vient pas du coeur.                Sa yéene ji àggul ci sa xol.
94                           Coupe le jusqu'au coeur.                   Gor ko ba àgg ca yeññ wa.
95  Va jeter un coup d'oeil pour voir s'il est rév...            Demal yër ba

- Identify translations with only a closing parentheses

In [59]:
scraped_sentences, re_sents = extract_with_re(scraped_sentences, regex='.*\).*')


You recuperated the following sentences:
                                               french                                              wolof
0   La marmite, si elle doit être délicieuse, quan...                Cin, su naree neex, bu baxee, xeeñ.
1        L'homme et ses désirs, Dieu et sa décision).              Nit ak bëgg-bëggam, Yàlla ak dogalam.
2   C'est avec l'esprit avec lequel tu as creusé l...     Xel mi nga gase teen bi la ci keneen di naane.
3   Une langue ne mange pas son semblable, si on n...           Làmmiñ du lekk morom ma, te toggeesu ko.
4   Si ton talent de cavalier te pousse à monter s...  Su la mën ngawar jayee ba nga mafñàndum saaw, ...
..                                                ...                                                ...
26                      Ton corps est-il en paix ?) .                            Sa yaram jàmm bonjour !
27  Généreux, avec tes biens) sois généreux, mais ...                               Yéwén : ci sa alal !
28  Qui a broy

Let us save the sentences and delete the space between a point or a comma and the word behind them. 

In [60]:
scraped_sentences.to_csv('data/extractions/new_data/scraped_sents.csv', index=False)

Let us add the scraped sentences inside a new collection and add it also the translations from the already existing collection.

In [39]:
scraped_sentences = pd.DataFrame(scraped_sentences)

database_manager.insert_documents([{
        '_id': id,
        'french': scraped_sentences.loc[id, 'french'],
        'wolof': scraped_sentences.loc[id, 'wolof'],
    } for id in scraped_sentences.index], 'sentences_plus')

Let us insert inside the collection the translations from the sentences collection.

In [40]:
hug_sents.index = range(scraped_sentences.shape[0], scraped_sentences.shape[0] + hug_sents.shape[0])
database_manager.insert_documents([{
        '_id': id,
        'french': hug_sents.loc[id, 'french'],
        'wolof': hug_sents.loc[id, 'wolof'],
    } for id in hug_sents.index], 'sentences_plus')

### Removing duplicated rows (advanced) and adding transformations

---------------

#### Without scraped sentences

Let us consider only the fourth end mark adding's method and verify if we have duplicated rows.

- Corpora v6

In [39]:
french_sents = corpora_v6['french'].to_list()

wolof_sents = corpora_v6['wolof'].to_list()

french_sents = add_end_mark(french_sents, end_mark_to_remove='!')

wolof_sents = add_end_mark(wolof_sents, end_mark_to_remove='!')

corpora_v6_end = pd.DataFrame({'french': french_sents, 'wolof': wolof_sents})

In [40]:
duplicated = corpora_v6_end.duplicated()

sum(duplicated)

5

We obtain 5 duplicated rows. Let us recuperate their indexes.

In [41]:
dup_indexes = duplicated[duplicated == True].index

In [42]:
dup_indexes

Int64Index([1426, 1595, 2239, 2241, 2365], dtype='int64')

Let us remove those indexes from the corpora.

In [43]:
corpora_v6.drop(index = dup_indexes, inplace=True)

- Diagne's sentences

In [44]:
french_sents = ad_sentences['french'].to_list()

wolof_sents = ad_sentences['wolof'].to_list()

french_sents = add_end_mark(french_sents, end_mark_to_remove='!')

wolof_sents = add_end_mark(wolof_sents, end_mark_to_remove='!')

diagne_end = pd.DataFrame({'french': french_sents, 'wolof': wolof_sents})

In [45]:
duplicated = diagne_end.duplicated()

sum(duplicated)

5

We obtain 5 duplicated rows. Let us recuperate their indexes.

In [46]:
dup_indexes = duplicated[duplicated == True].index

In [47]:
dup_indexes

Int64Index([544, 713, 1357, 1359, 1483], dtype='int64')

Let us remove those indexes from the corpora.

In [48]:
ad_sentences.drop(index = dup_indexes, inplace=True)

----------------------------

#### With scraped sentences and sampling

Let us add the scraped sentences into the different datasets and consider only 5000 samples with the random state 0.

In [59]:
shape_1 = ad_sentences.shape[0]

In [60]:
# inside diagne's sentences
add_sentences = pd.concat((ad_sentences, pd.DataFrame(scraped_sentences))).sample(5000, random_state = 0)

# inside the corpora
corpora_v7 = pd.concat((corpora_v6, pd.DataFrame(scraped_sentences))).sample(5000, random_state = 0)

# inside diagne's sentences (we want 5000 translations for diagne's sentences)
# ad_sentences = pd.concat((ad_sentences, pd.DataFrame(scraped_sentences).sample(3000 - shape_1)))


In [61]:
add_sentences.index = range(add_sentences.shape[0])
corpora_v7.index = range(corpora_v7.shape[0])

Let us consider only the fourth end mark adding's method and verify if we have duplicated rows.

- Corpora v7

In [62]:
french_sents = corpora_v7['french'].to_list()

wolof_sents = corpora_v7['wolof'].to_list()

french_sents = add_end_mark(french_sents, end_mark_to_remove='!')

wolof_sents = add_end_mark(wolof_sents, end_mark_to_remove='!')

corpora_v7_end = pd.DataFrame({'french': french_sents, 'wolof': wolof_sents})

In [63]:
duplicated = corpora_v7_end.duplicated()

sum(duplicated)

4

We obtain 5 duplicated rows. Let us recuperate their indexes.

In [64]:
dup_indexes = duplicated[duplicated == True].index

In [65]:
dup_indexes

Int64Index([1108, 1850, 3641, 4398], dtype='int64')

Let us remove those indexes from the corpora.

In [66]:
corpora_v7.drop(index = dup_indexes, inplace=True)

- Diagne's sentences

In [67]:
french_sents = add_sentences['french'].to_list()

wolof_sents = add_sentences['wolof'].to_list()

french_sents = add_end_mark(french_sents, end_mark_to_remove='!')

wolof_sents = add_end_mark(wolof_sents, end_mark_to_remove='!')

diagne_end = pd.DataFrame({'french': french_sents, 'wolof': wolof_sents})

In [68]:
duplicated = diagne_end.duplicated()

sum(duplicated)

4

We obtain 5 duplicated rows. Let us recuperate their indexes.

In [69]:
dup_indexes = duplicated[duplicated == True].index

In [70]:
dup_indexes

Int64Index([2686, 3248, 3416, 4316], dtype='int64')

Let us remove those indexes from the corpora.

In [71]:
add_sentences.drop(index = dup_indexes, inplace=True)

-------------------

Apply transformations on all sentences.

In [75]:
# apply the transformations to all the sentences
def correction(text: str):
    
    text = transformer(text)[0]
    
    return text

# add corrections to the translations
add_sentences['french'] = add_sentences['french'].apply(correction)
add_sentences['wolof'] = add_sentences['wolof'].apply(correction)

corpora_v7['french'] = corpora_v7['french'].apply(correction)
corpora_v7['wolof'] = corpora_v7['wolof'].apply(correction)

ad_sentences['french'] = ad_sentences['french'].apply(correction)
ad_sentences['wolof'] = ad_sentences['wolof'].apply(correction)

corpora_v6['french'] = corpora_v6['french'].apply(correction)
corpora_v6['wolof'] = corpora_v6['wolof'].apply(correction)

diagne_sentences['french'] = diagne_sentences['french'].apply(correction)
diagne_sentences['wolof'] = diagne_sentences['french'].apply(correction)

corpora_v4['french'] = corpora_v4['french'].apply(correction)
corpora_v4['wolof'] = corpora_v4['french'].apply(correction)

# print clearly the results
ad_sentences.sample(74)


french  \
1172                                        Oui Samba !   
1122                         C'est peut-être celui-là !   
623                               Ce n'est pas l'autre.   
69    Ceci est un lieu d'habitation très charmant, t...   
4                               Comment va ta famille ?   
523                                  Les voilà, là-bas.   
1254                            J'ai vu cet enfant-là ?   
880                Tu es paresseuse comme cet autre-là.   
360                 Comme il avait ainsi été en ce lieu   
116   Sur la photo que vous m'avez envoyée, j'y ai v...   
974            Il a acheté moutons, chèvres, poulets...   
830             Il a donné à la personne quelque chose.   
1250                        Je parle de cet enfant-ci ?   
1447                                Donne un à chacun !   
962                    L'homme et l'enfant sont partis.   
188                                 C'est leurs dents !   
227                                      Si tu es homme   
1285                              Attache tout mouton !   
44                            Tu avais vu cette femme ?   
1505     Je l'ai donné à quelqu'un de meilleur que lui.   
1151                             Le voilà et le voilà !   
6                                               Pardon.   
5     Bonjour ! Ceci est une photo. Sur cette photo ...   
61                                           Vas t'en !   
509                                       Toi tu as été   
38                Cause avec cet homme qui est à côté !   
232                      Sois l'homme de la situation !   
302                                       Ne partez pas   
106                          Va voir tous les moutons !   
477                 C'est ceux-là qui n'eussent pas été   
23                                            Bon matin   
13                                             Et toi ?   
695                         Et tu n'as été nulle part ?   
1300                         Tu parles de quel mouton ?   
1282                Quels moutons qui t'appartiennent ?   
1569     Que l'homme parte et que les enfants entrent !   
904                  Il y amena un mouton pour l'homme.   
210                                  Nous sommes partis   
844                          Je ne le lui donnerai pas.   
511                           L'homme a parlé avec toi.   
644                                     Tu as dit cela.   
1263                             J'ai aperçu un cheval.   
187          Le U2 a pris une tonne de film sous l'eau.   
1383                 Ceux-ci ne partent peut-être pas !   
717                                    C'est un boabab.   
1198                             Quel homme est parti ?   
137                            Cherche une autre dame !   
251                    Tout village Sérère est propre !   
97    Ce que j'ai vu c'est une pierre à feu. Je croi...   
1583  Les enfants veulent venir et les adultes souha...   
383                                 L'homme n'a pas été   
951                   La vieille, elle, ne parla plus !   
1404              Tu aurais conversé avec qui d'autre ?   
655                                 De quelle manière ?   
10                              Comment t'appelles-tu ?   
1398  Aucun de tes neveux n'est aussi paresseux que ...   
537                         Dis à l'homme qu'il parte !   
1174                               À l'intérieur même ?   
6                                 Quelqu'un est parti !   
657                               Tout cet endroit-là ?   
23                                Regarde cette aumône.   
65                                           À l'aide !   
85                            Quels amis sont arrivés ?   
1240  Cet homme-ci, devant nous, et cet homme-là ne ...   
731                                   Que pleure-t-il ?   
930                        Cela seulement ne suffit pas   
1463                   L'individu respire avec son nez.   
1599  L'homme est parti pour que vous

In [76]:
corpora_v6.sample(74)

french  \
1982                     Ici, jusqu'à ce qu'il vienne !   
1609                               Mais NGor est parti.   
227   Il me semble qu'il y avait dans ce rituel du t...   
883                        Je t'entrevois, seul, emmuré   
611   Si c'est toujours un peu l'aventure (on est to...   
423   J'imagine son exaltation à l'arrivée à Victori...   
2429        Chez les Peul, les Sérère, et les Niominka.   
13    Ceci est un ancien milieu avec une constructio...   
201        L'augmentation du trafic aérien est gênante.   
70                               Mes voeux de nouvel an   
47    L'Afrique qui déjà m'ôtait mon visage me renda...   
59    Un jardin, soit, puisqu'un des employés de la ...   
176   Nous étions venus d'un appartement au sixième ...   
1407                      Le voilà, là-bas, qui s'en va   
1211                                          Dites-lui   
96    Les souvenirs trompent, sans doute. Cette vie ...   
12    Sur cette photo, il y a d'une salle dans la qu...   
150   Sur la photo que vous m'avez envoyée, j'ai vu ...   
1954                        C'est l'homme qui est venu.   
1081                           Tu as reconnu sa femme ?   
664                            Il était dur, taciturne.   
2                                                 Salut   
1633                           Qui t'a donné ce livre ?   
2485  Allez pour que l'arachide arrive et qu'enfin l...   
186                      Le U2 a pris beaucoup de film.   
298   Je me souviens aussi d'avoir été pris par des ...   
785   J'écouterai la voix de ma mère qui compte les ...   
715                 Tout cela peut sembler anecdotique.   
558   Un peu avant le départ de mon père, Jeffries t...   
1351                 Tu ne bavardais pas, généralement.   
38    Parmi tous ceux qui se pressent autour de moi,...   
330   Mon père et elle sont unis par ce rêve, ils so...   
391   À son retour en France, il avait gardé les hab...   
1853             Toi ou lui, il en est un qui est parti   
1191                      C'est lorsque la femme arriva   
1340                      L'homme n'est pas un orfèvre.   
2468  Les enfants veulent venir et les adultes souha...   
2008                        C'est celui-là, peut-être !   
689   Où la maladie et la vieillesse étaient visible...   
2123                   Tu reconnais cette personne-là ?   
2176                          L'un des hommes est venu.   
1195                          L'homme qui eût travaillé   
2206                          Elles ont été quelqu'un !   
7                              Paix seulement, et toi ?   
1419             Sur ces entrefaites, moi, je quittais.   
2040                           C'est un seul, un seul !   
455   Et avant ma vie, ils parlaient du temps que mo...   
35    Mon père a arrêté son auto, avec ma mère il do...   
730   Il est mort l'année où le sida a fait son appa...   
1443                                     Tu as vu qui ?   
1836                  La vieille, elle, ne parla plus !   
1373                                           J'ai été   
2272  Des boeufs que tu vois dans cette ville, le pl...   
1263                               Mais l'homme cultive   
2231                                       Nous partons   
1761                   Il est mal élevé comme un gosse.   
512   Ils y restent une ou deux nuits, parfois une s...   
1735                                 Un lézard par là !   
385   Ses malles de bois cerclées de fer, dont il av...   
1043                                  C'est votre ami !   
14                                      Mon nom est ...   
22                              Ils se tiennent dehors.   
1824                          C'est là ce que je veux !   
250   Mon père est arrivé en Afrique en 1928, après ...   
1623                                     Qui as-tu vu ?   
526   Sur la place, le théâtre masqué se prépare. So...   
280   La première fois que j'ai vu mon père, à Ogoja...   
183                     Tous leurs en

In [74]:
add_sentences.sample(74)

french  \
0   Quand j'étais enfant, il y avait un figuier à ...   
1                                            Et moi ?   
2                                        Dis-le-lui !   
3                       Il l'a attrapé par la taille.   
4   Je suis passé par deux fois devant la maison, ...   
5                                         Le pauvre !   
6                                   Un enfant retors.   
7                       Où sont les nouveaux mariés ?   
8                                Le soleil est sorti.   
9    Ce qui germe dans le coeur se reflète au visage.   
10  Un adulte qui fait de la pédophilie, je sais v...   
11  Chaque fois qu'arrive la saison des pluies, sa...   
12               Je n'aime pas les aubergines frites.   
13                 Enfume le trou et le rat sortira !   
14                                   Un boubou jaune.   
15  C'est par les crânes qu'on détermine le nombre...   
16                         Attrapez-moi la poulette !   
17                            Il manque une cuillère.   
18  Le plus souvent, ce sont les Peuls qui transhu...   
19               Je vous invite dimanche à la maison.   
20               Il est « excessivement » courageux !   
21  Est-ce que tu peux me procurer un peu de charb...   
22                             Nettoie l'avant-bras !   
23          Elle s'est assise fermement sur la natte.   
24        Laissez ce jeu de aayet allez à la cuisine.   
25            J'ai laissé un message pour toi à Omar.   
26                      Une plume n'est qu'une plume.   
27                         Tu connais cet enfant-ci ?   
28  Quand elle remplissait de lait celui de son fi...   
29        L'homme et ses désirs, Dieu et sa décision.   
30  Certaines choses, il faut les avoir expériment...   
31                 Le bracelet est serré, écarte-le !   
32  Si tu veux quelque chose à aller raconter, tu ...   
33      C'est la mangue qui a attiré les mouches ici.   
34  Les provisions que tu as faites ne suffiront p...   
35                                          Il tonne.   
36                          L'homme qui eût travaillé   
37    On croyait que tu allais partir matin et soir !   
38                                    Il est aveugle.   
39             Donne la gomme arabique à la lingère !   
40      J'ai apporté aux enfants de la canne à sucre.   
41  Quand les relations entre les deux pays se son...   
42                                       Nous fumons.   
43  Oh là petit bonhomme, où t'en vas-tu à cette h...   
44                Un jour, le lion surgit devant lui.   
45                 Ce qui avait cours n'a plus cours.   
46  Dans un pays, quand il arrive que la reine dan...   
47              Astou a quitté son domicile conjugal.   
48  Va réviser le verset, tu ne l'as pas encore bi...   
49           La pipe que tu lui as offerte est belle.   
50                         Les dartres t'ont enlaidi.   
51  Seigneur, prends pitié de moi car je ne suis q...   
52                            Où est le surveillant ?   
53  Il a emprunté ma chemise et en a fait sa propr...   
54  Quel Modou ? Ne s'agirait-il pas de Modou Pouye ?   
55    Je n'aime pas la purée d'oseille avec du gombo.   
56  Décroche la chemise, mets-la avec le linge à l...   
57  L'oiseau s'est perché sur le faîte pointu du t...   
58  Attends que la terre soit suffisamment imbibée...   
59  Leur balle est tombée dans le tas de paille qu...   
60  Passe encore que tu ne fasses pas l'aumône à u...   
61                     Toi tu eusses été notre guide.   
62                       Qui as-tu envoyé au marché ?   
63           Ton feu est bien pris ; viens commencer.   
64                     Il n'en a fait qu'une bouchée.   
65  Celui qui sert le repas avec la main, sans la ...   
66  Si tu laves le poisson là où tu l'as pris, il ...   
67                                Ne le décolle pas !   
68                  Ils marchent le long de la plage.   
69        Vois quelqu'un qui est libre, va avec lui !   


In [77]:
corpora_v7.sample(74)

french  \
4889                           Rien ne va mal avec eux.   
2550  Pendant la disette, on prenait des pépins de p...   
607                Remets le couteau dans le fourreau !   
3939                     Cela a coïncidé avec ma venue.   
1654  Les chiens sortirent dans tous les sens, le vi...   
4354             Tu n'es pas aussi modeste que ta mère.   
11                    C'est ça que j'avais vu chez toi.   
2691  C'est dans cette clairière qu'on a organisé le...   
1027                                Deux jours au plus.   
1471  Va te laver la figure pour que les traces de b...   
3350   Maintenant, il fait des efforts dans ses études.   
4515                                   Sables mouvants.   
2235                          La vache lèche son petit.   
4573                Les routes étaient en mauvais état.   
3304  Je sais que cela m'a manqué, sans regret, sans...   
2081                                   Il s'arrêta net.   
1828                             Tu vois cet homme là ?   
1983  La maison de fonction qu'il habite est juste à...   
3045                 Le cheval est en train de galoper.   
2754  Après l'avoir délayé, tu le laisses reposer, t...   
4343  C'est là qu'il va travailler pendant les premi...   
2252            Il ne sait que dalle, il n'a que dalle.   
4660                     Ta sauce est bien assaisonnée.   
3456                                  Il a la diarrhée.   
2034  Pendant longtemps j'ai polémiqué, pendant long...   
367                    Que la paix soit avec toi aussi.   
1768             Ne te lasse pas, ne te décourage pas !   
2875                              Voici toute la chose.   
2217   C'est tout ce qui est beau qui séduit cet homme.   
3670                                    je te briserai.   
4456                     En résumé : il ne partira pas.   
4787                   Vers ceux qui sont debout là-bas   
541                 Est-ce que ton père est de retour ?   
3029                        J'ai paré le coup de poing.   
1839              Il se débattait pour aller se battre.   
2353  Chaque fois que je te rappelle la chaîne que t...   
1814  J'enseignement sera difficile pour lui car il ...   
1377                           Qui allume le fourneau ?   
3001                          Ces enfants sont impolis.   
996   Alors, il n'y avait pas de voitures, on allait...   
4612                                  C'est un albinos.   
4906              C'est Omar qui a hérité de la maison.   
1033  Les béliers châtrés sont gras la plupart du te...   
1877                 C'est ici qu'ils les affrontèrent.   
4296                   Il a dit à Moussa quelque chose.   
3715  Là-bas, de l'autre côté de la mer, le monde s'...   
4835  Vraiment, nous n'avons pas le même point de vu...   
2406                          Cet homme qui était parti   
532   Et les noms de pays, Mbembé, Kaka, Nsungli, Bu...   
3293                                 Il n'est pas futé.   
3968      Tu n'es pas encore né, et tu as la tremblote.   
2035                                 Ce serpent crache.   
3338    On leur enseigne la meilleure façon d'allaiter.   
2656                Oriente le chevet vers la fenêtre !   
440   Allons dans la boutique nous abriter de la plu...   
1206                             J'ai donné un exemple.   
1976  Manger du pain avec de l'eau sucrée est la sol...   
3570            Attends que je soulève le pied du lit !   
1645                         Je veux clôturer le champ.   
3235             Tu lui élimineras les poux de la tête.   
2547                      Ne fais pas monter sa colère.   
2540                            J'ai vu cet enfant-là ?   
4976                                   Vous êtes partis   
1332              Je ne suis jamais allé à cet endroit.   
833                                    Pour qui es-tu ?   
1496                           Pour un peu, il tombait.   
1086                                         Je t'aimes   
51                           Poussent

Let us save the results.

In [78]:
# save the results
add_sentences.to_csv('data/extractions/new_data/add_sentences.csv', index=False)
corpora_v7.to_csv('data/extractions/new_data/corpora_v7.csv', index=False)
ad_sentences.to_csv('data/extractions/new_data/ad_sentences.csv', index=False)
corpora_v6.to_csv('data/extractions/new_data/corpora_v6.csv', index=False)
diagne_sentences.to_csv('data/extractions/new_data/diagne_sentences.csv', index=False)
corpora_v4.to_csv('data/extractions/new_data/corpora_v4.csv', index=False)

Let us verify how many sentences we got for the moment.

In [79]:
# from diagne's book without scraped sentences
ad_sentences.shape[0]

1977

In [80]:
# inside the corpora without scraped sentences
corpora_v6.shape[0]

2859

In [81]:
# from diagne's book with scraped sentences
add_sentences.shape[0]

4996

In [82]:
# inside the corpora with scraped sentences
corpora_v7.shape[0]

4996

~~We recuperated `11435` sentences from the diagne's book and the final corpora contains `12317` sentences.~~